src/test/resources/org/jenkinsci/plugins/ewm/casc/configuration-as-code.yaml

```diff
@@ -1,14 +1,58 @@
 jenkins:
-  systemMessage: "Jenkins configured automatically by Jenkins Configuration as Code Plugin\n\n"
+  systemMessage: "Jenkins configured automatically by Jenkins Configuration as Code Plugin"
   numExecutors: 5
   scmCheckoutRetryCount: 2
   mode: NORMAL
+  nodeProperties:
+    - exwsNodeConfigurationDiskPools:
+        nodeDiskPools:
+          - diskPoolRefId: "master-node-id"
+            nodeDisks:
+              - diskRefId: "master-node-disk"
+                nodeMountPoint: "/tmp/master-node"
+
+# TODO [Martin]: The following code crashes JCasC for reasons I do not understand
```

Comment:
Anyone knows what is wrong here?

Reply:
Could it be that the expected singleton is not a singleton?

Reply:
I get this error when I comment out that code:
```
[ERROR] Tests run: 2, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 5.689 s <<< FAILURE! - in org.jenkinsci.plugins.ewm.casc.ConfigAsCodeTest
[ERROR] shouldSupportConfigurationAsCode(org.jenkinsci.plugins.ewm.casc.ConfigAsCodeTest)  Time elapsed: 0.047 s  <<< ERROR!
java.lang.NullPointerException
        at io.jenkins.plugins.casc.model.Mapping.lambda$clone$0(Mapping.java:71)
        at java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1691)
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
        at io.jenkins.plugins.casc.model.Mapping.clone(Mapping.java:71)
        at io.jenkins.plugins.casc.model.Mapping.clone(Mapping.java:11)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
        at io.jenkins.plugins.casc.model.Sequence.clone(Sequence.java:43)
        at io.jenkins.plugins.casc.model.Sequence.clone(Sequence.java:9)
        at io.jenkins.plugins.casc.model.Mapping.lambda$clone$0(Mapping.java:71)
        at java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1691)
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
        at io.jenkins.plugins.casc.model.Mapping.clone(Mapping.java:71)
        at io.jenkins.plugins.casc.model.Mapping.clone(Mapping.java:11)
        at io.jenkins.plugins.casc.model.Mapping.lambda$clone$0(Mapping.java:71)
        at java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1691)
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
        at io.jenkins.plugins.casc.model.Mapping.clone(Mapping.java:71)
        at io.jenkins.plugins.casc.ConfigurationAsCode.configureWith(ConfigurationAsCode.java:641)
        at io.jenkins.plugins.casc.ConfigurationAsCode.configureWith(ConfigurationAsCode.java:545)
        at io.jenkins.plugins.casc.ConfigurationAsCode.configure(ConfigurationAsCode.java:520)
        at io.jenkins.plugins.casc.ConfigurationAsCode.configure(ConfigurationAsCode.java:503)
        at org.jenkinsci.plugins.ewm.casc.ConfigAsCodeTest.shouldSupportConfigurationAsCode(ConfigAsCodeTest.java:43)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:548)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.lang.Thread.run(Thread.java:748)

[ERROR] exportConfiguration(org.jenkinsci.plugins.ewm.casc.ConfigAsCodeTest)  Time elapsed: 0.272 s  <<< FAILURE!
java.lang.AssertionError: expected:<{exwsGlobalConfigurationDiskPools={diskPools=[{diskPoolId=diskpool1, disks=[{diskId=disk1, displayName=disk one display name, masterMountPoint=/tmp}], displayName=diskpool1 display name}]}, exwsGlobalConfigurationTemplates={templates=[{label=all, nodeDiskPools=[{diskPoolRefId=dp1, nodeDisks=[{diskRefId=dp1refid1, nodeMountPoint=/tmp/template11}, {diskRefId=dp1refid2, nodeMountPoint=/tmp/template12}]}, {diskPoolRefId=dp2, nodeDisks=[{diskRefId=dp2refid1, nodeMountPoint=/tmp/template21}]}]}]}}> but was:<null>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.jenkinsci.plugins.ewm.casc.ConfigAsCodeTest.exportConfiguration(ConfigAsCodeTest.java:93)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:548)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.lang.Thread.run(Thread.java:748)
```

Reply:
This would take a while for me to look at these stuffs. And my local repository cannot run local tests due to some unexpected error. (And that's why i didn't follow this PR after that break down ..... LOL )

Reply:
I think it should be enough if you define the node as:

```
  nodes:
    - dumb:
        mode: NORMAL
        name: "agent1"
```

However I'm still trying to figure out how to define the `nodeProperties` 😕